From 0a5a4c046d10c0a53c7d2120a523cc28663ad73b Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Tue, 10 Aug 2021 11:23:48 +0200
Subject: [PATCH] alsa: Free global state

We need to do this or else newly plugged devices might not load.

It does not seem to harm UCM config on my machine, so this reverts
3d372424ccb96fd2b705c9ed94e76a82ad44e113

See #1478
---
 spa/plugins/alsa/alsa-pcm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/spa/plugins/alsa/alsa-pcm.c b/spa/plugins/alsa/alsa-pcm.c
index 30fbbba6e..299bf03ff 100644
--- a/spa/plugins/alsa/alsa-pcm.c
+++ b/spa/plugins/alsa/alsa-pcm.c
@@ -20,6 +20,8 @@ int spa_alsa_init(struct state *state)
 {
 	int err;
 
+	snd_config_update_free_global();
+
 	if (state->open_ucm) {
 		char card_name[64];
 		const char *alibpref = NULL;
-- 
GitLab

